src/motor_hardware.cc

```diff
@@ -63,7 +64,7 @@ int32_t  g_odomEvent = 0;
 // This utility opens and reads 1 or more bytes from a device on an I2C bus
 // This method was taken on it's own from a big I2C class we may choose to use later
 static int i2c_BufferRead(const char *i2cDevFile, uint8_t i2cAddr, 
-                          uint8_t* pBuffer, int16_t chipRegAddr, uint16_t NumByteToRead);
+                          uint8_t* pBuffer, int16_t chipRegAddr);
 
```

Comment:
Why are we loosing the ability to read more than one byte?   Is that a restriction of this method of reading?

Reply:
Well previously we had two methods write() and read() in order to read from the chips register. And it was a must to check whether the number of bytes we requested then actually received, because of race condition. And it was straight forward to check it, because the read() method returned the number of bytes read. Now there is only one call ioctl() that combines both of the methods(write() and read()) and locks any other device that is trying to interfere in the meantime. Since ioctl() is the only call in this case, if the call was successful it returns a non-negative number, while if the call is unsuccessful it returns the -1.

Reply:
Yes I get the reason for the ioctl.  I once wrote a complex fibre channel driver for a unix like system that had it's reads and writes accessed using an IOCTL.   So I get that and I am glad we are going to use this interface.   My point was this routine should not be crippled to only read one byte but if you feel that is necessary then change it's name to  indicate it only reads one byte and change assorted comments in the routine that discuss number of bytes such as  'reads 1 or more bytes from a device on an I2C bus'  no longer applies and the header to the routine should also say it is only for reading one byte from a device with no internal read register address.  In other words I was hoping for this call to remain general purpose because we have plans to use this call in other places so getting it general purpose now helps for our plans to use this code in other places.    I hope this better explains the intent and am sorry if you got the wrong message earlier on this task.